wmr-api/controllers/sales.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+
+from odoo import http, models
+from odoo.http import request
+from werkzeug.exceptions import BadRequest
+import html2text
+import re
+
+
+# Authenticate the api request using the odoo api keys
+class IrHttp(models.AbstractModel):
```

Comment:
Does this need to be setup twice? Or could we create a separate file for the auth stuff and use it in all of the locations needed.